src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java

```diff
@@ -109,12 +112,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
    * Returns the shopping query by calling methods from the photo content detection classes, 
    * based on the {@code photoCategory}, passing {@code uploadedImageBytes} as argument.
    */
-  private String getQuery(String photoCategory, byte[] uploadedImageBytes) throws IllegalArgumentException {
+  private String getQuery(String photoCategory, byte[] uploadedImageBytes)
+      throws IOException, PhotoDetectionException {    
```

Comment:
There's also the IllegalArgumentException here :)
```suggestion
      throws IllegalArgumentException, IOException, PhotoDetectionException {    
```